spec/models/supermarket_spec.rb

```diff
@@ -4,4 +4,21 @@
   describe 'relationships' do
     it { should have_many :items }
   end
+
+  describe 'instance methods' do
+    describe '#unique_customers' do
+      it 'returns a unique list of customers for that supermarket' do
+        will = Customer.create!(name: 'Will')
+        stephen = Customer.create!(name: 'Stephen')
+        soopers = Supermarket.create!(name: 'Soopers', location: 'Denver')
+        steak = soopers.items.create!(name: 'Steak', price: 10)
+
+        will.items << steak
+        will.items << steak
+        stephen.items << steak
+
+        expect(soopers.unique_customers).to eq([will, stephen])
```

Comment:
I would add a customer that does not shop at soopers just to ensure that `unique_customers` is working as you intend.